### ISSUE TITLE
Invisible hostile mobs.

### ISSUE THREAD

**Issue by Quavelen** [sp614x/optifine#1892]
## Edit as of 31 January 2021:

So, I've started poking around in Minecraft again, and this issue is still occurring in 1.16.5 for me. I exclusively play on servers (running the 'Paper' server software), and this is where I encounter invisible mobs.

According to the 72 (wow! o.o) comments on here, it seems this never stopped being an issue.

Right now I'm using these:

![image](https://user-images.githubusercontent.com/19168008/106397787-16807c80-6407-11eb-86ee-bfe2d3c0c2ea.png)

The issue is still as infrequent as ever, and annoying to reproduce. I'm *currently* unable to intentionally make it happen, but it happens when I really don't want it to... funny how the world works like that, lol.

Looks like there's another large thread (#5006) from 6 November 2020, where people are experiencing it as well.

______

## Edit as of 03 August 2020:

I have not actively played Minecraft since I reported this issue, but based on the infrequent comments which keep reconfirming it, it's likely still an issue. 

Please read through all the comments on this before replying.
If at all possible, try to provide the same information sp614x asked of me when/if you do comment. **This is important.**

Since I am inactive on this, it may even be worth opening your own issue for it. I'm not really sure what sp614x's policy on this is.

_______

## Original submission on 12 October 2018:

This has affected all versions of the 1.13.1 previews, from alpha, through beta, and now in pre5. I am not the only one experiencing it, as others have asked me if it has been fixed yet.

Invisible zombie in OptiFine 1.13.1 pre5, with it attacking me and subtitles showing it's there: https://giant.gfycat.com/NeglectedCarefulJohndory.webm

Visible zombie in vanilla; same exact spot: https://giant.gfycat.com/ClutteredMildGrizzlybear.webm

The bug is random, and sometimes mobs are visible, other times they're not. In vanilla, it never occurs and all mobs are always visible. This is also not limited to servers; it happens in single player, and only with OptiFine. In my opinion, this is a rather game-breaking bug since I almost exclusively play Survival.

All needed information can be seen in both of the above webms, I have my F3 menu open, which displays hardware, game version, etc.

Side note: If you'd rather I use a different file host for the two webms (like YouTube) let me know. I know that not everyone is aware of Gfycat, so it can look like a sketchy link.

**Comment by sp614x** [comment]
Resource packs can use Random Entities to assign custom textures to mobs. They can also assign partially or fully transparent textures to mobs. For example to simulate missing limbs or invisible mobs.
Vanilla doesn't recognize the Random Entities configuration, so it always shows the default skin.

**Comment by Quavelen** [comment]
Yes, sorry @sp614x. It happens without any resource packs as well. Here's one I recorded without one enabled: https://giant.gfycat.com/AlarmingEmptyArthropods.webm

To force it to happen, I can just repeatedly log out and back in, or '/spawn' and '/back' until it turns invisible. For single player, I can just leave and exit the world or leave the chunks it's in and come back. It may take a few tries, but it will happen.

**Comment by Quavelen** [comment]
Okay, @sp614x. Here's a proper video showing that it still occurs in the full release - OptiFine_1.13.1_HD_U_E3: 

https://www.youtube.com/watch?v=HvabxTuvVJQ

Again, no texture packs whatsoever. I open the texture pack menu to prove this, then logout/login a few times to attempt to make it become visible. Ended up having to just unload the chunks and then return. If it were related to the "random mobs" feature, then this would not happen in vanilla textures. Unless I'm mistaken, and I have them enabled somehow regardless.


Now all this said - I **_never_** get random invisible hostiles like this without OptiFine. As soon as I stop using it, hostiles are always visible at all times. Even with texture packs. So, please tell me how to fix this if you are not experiencing it, because it is a game breaking issue in survival.

**Comment by mattebad** [comment]
Hey I just wanted to piggy back off what @Quavelen said. I'm experiencing the same thing, at first it was just partial mob invisibility, mostly only on spiders. Just updated to the full release of optifine for 1.13 and I'm experiencing nearly 90% invisible mobs. Turning shaders off and restarting. Same deal. Turning both resource pack and shaders off (Chroma Hills, and Continuum). Same deal. Mobs fully invisible.

I waited for morning to roll around and I couldn't even see them burn, but could hear them burning and saw them drop loot. The server I'm playing on even has a health bar plugin that shows entity health, and none of the mobs I heard near me had health bars on them which is quite weird, even though they're invisible I would have thought that would have worked.

I just want to confirm that yes, this is a issue, and it started from updating from pre-7 to the full release of 1.13

**Comment by sp614x** [comment]
Some errors in error log or the game output?
Can you post steps to reproduce it, for example: new world seed ABC, go to X,Y,Z, spawn zombie, etc.?

**Comment by mattebad** [comment]
As far as I can tell in terms of reproducing it, it occurs at random (First time start after updating to 1.13 optifine). But when it does, it stays like that even after a full restart of the launcher and minecraft itself multiple times. Only after restarting and rejoining the server multiple times, as well as waiting for a full day, will it fix itself somehow. This is on a friends server that has had no issues, as well as locally in my world. From what I can figure out, the mobs don't seem to be physically loaded into the game as no health shows up, nor am I able to hit/damage the mob, but it can damage me. So I don't think it's a texture issue, otherwise, I could damage the mob or see it's health. 

Even more weird, is that, waiting a full day has the mobs killed off, and when they respawn during the next night, they work fine consistently and I don't have any issues following that.

**Comment by Quavelen** [comment]
@sp614x it's exactly as @mattebad described. There is nothing in the launcher log/game output to imply it's an error. The mobs - when they turn invisible - are completely impervious. You cannot hit them, even if you trap them and repeatedly attack the same spot. I actually showed you this in one of the above webms above; you can see me attacking the area where the zombie is trapped. 

This issue is entirely random, and seems to have no way to reproduce or fix other than repetitively logging out/in, and loading/unloading the chunks. Either will work, but it takes a few attempts.

**Comment by sp614x** [comment]
Does it depend on Random Entities -> ON / OFF ?

**Comment by Quavelen** [comment]
@sp614x Just tested that for you, with it off, and no. They still turn invisible. All previous attempts were with it turned on.

**Comment by sp614x** [comment]
May be related: https://www.reddit.com/r/Minecraft/comments/9plwrn/optifine_buggy/

**Comment by abungato** [comment]
I am also experiencing this issue when using the latest OptiFine install. Zombie and spider mobs are invisible and I cannot attack them.

**Comment by peaktogoo** [comment]
Still happening in Lastest optifine, & lastest pre version

Invisible Phantom hitted me with only sound ;_;

**Comment by Quavelen** [comment]
Likewise, still happening on the 1.13.2 pre2 for me.

**Comment by sp614x** [comment]
Looks like the mobs are not loaded client-side at all, they only exist in the (integrated) server.
This is why they can't be hit (hit detection is client-side), but the player can be damaged (server-side check) and they produce noises (server initiated).

**Comment by Quavelen** [comment]
> Looks like the mobs are not loaded client-side at all, they only exist in the (integrated) server.
> This is why they can't be hit (hit detection is client-side), but the player can be damaged (server-side check) and they produce noises (server initiated).

Yay! Does this mean you know how to fix it? I'm really looking forwards to not having random invisible mobs any more, lol.

**Comment by sp614x** [comment]
With render distance 2, are the invisible mobs shown in the debug info as entities?

![image](https://user-images.githubusercontent.com/16635441/47569600-49885b80-d934-11e8-8ff9-5650cec267a1.png)

**Comment by Quavelen** [comment]
Nope, it shows as zero:

![invisible](https://i.imgur.com/zhjwer4.png)

![visible](https://i.imgur.com/p7k4EO9.png)

**Comment by sp614x** [comment]
Can you repeat the same test with full debug screen (F3)?

**Comment by Quavelen** [comment]
Yup, here you go:

![invisible](https://i.imgur.com/WGTdx7G.png)

![visible](https://i.imgur.com/4otizGi.png)

**Comment by sp614x** [comment]
The total number of entities is different 245 vs 368.

![image](https://user-images.githubusercontent.com/16635441/47571192-42fbe300-d938-11e8-9c7a-0ee62663e672.png)
![image](https://user-images.githubusercontent.com/16635441/47571208-4abb8780-d938-11e8-975e-d6eca4ce8b4b.png)

This could mean that the entities are not skipped when rendered, but really not loaded.

**Comment by Quavelen** [comment]
Well, if I correctly understand how Spigot works, un-name tagged hostile mobs are despawned when the chunks are unloaded. I was repeatedly using '/back' to teleport between spawn chunks, and the area where that named zombie was held. So that number could simply be the difference between mobs that were despawned when I teleported.

**Comment by sp614x** [comment]
These are the client total entities. They can only exist in chunks loaded by the client, which are limited by the render distance. The server keep more entities loaded, but they shouldn't be sent to the client when they are outside the chunk render distance.

**Comment by Quavelen** [comment]
Is this (invisible mobs) something you can fix, though? I don't understand how the stuff works; I just know it only happens with OptiFine.

**Comment by sp614x** [comment]
So far I can't reproduce the invisible mobs. No amount of load, unload, teleport, etc. can create unvisible mobs. The problem can be fixed if I can reproduce it and see exacly what happens.

**Comment by gitfamhub** [comment]
I'm getting this on 1.12.2 as well.

**Comment by IntelMiner** [comment]
I'm getting this on 1.13.2-OptiFIne_HD_U_E4 as well

**Comment by elkala** [comment]
I have the same problem, it happened to me with a zombie and a spider, same game and optifine version as @IntelMiner

**Comment by sp614x** [comment]
This can only be fixed if it is reproducible. Like "make a new world with seed ABC, at coord XYZ there is an invisible zombie" or something similar.

**Comment by Quavelen** [comment]
Well @sp614x, could you try testing on a spigot server? That's where I mainly see them. Since opening this issue, I haven't been able to get invisible mobs in single player; only with OptiFine on a Spigot server. This is despite vividly recalling them being invisible in single player.


Just to reiterate:
* Without OptiFine, all mobs are always perfectly visible regardless of what I do.
* With OptiFine, mobs are occasionally invisible upon loading chunks.
  * Loading chunks includes joining server, and teleporting; haven't tried F3+A.
  * Random Entities off/on has no effect; happens with either.
  * Texture packs/shaders have no effect; happens with either, and default.

I'm also not sure if this requires certain plugins, or if it can just be Spigot by itself (or other server software). I will try spigot by itself right now, though.

**Comment by sp614x** [comment]
Probably related: https://bugs.mojang.com/browse/MC-65040

**Comment by ScionNinja14** [comment]
Same issue here. Once I found the invisible mob, I relaunched Minecraft without optifine and then the zombie was actually visible, and I could interact with it. 
To make sure simply relaunching didn't fix the issue, I went and found another invisible mob and then relaunched with optifine and the mob was still invisible. 
Is there any fix for this soon? Getting real annoying dying to invisible creepers.

**Comment by sp614x** [comment]
@ScionNinja14 single-player or multiplayer (which server)?

**Comment by ScionNinja14** [comment]
@sp614x its in a spigot multiplayer.

**Comment by sp614x** [comment]
Is there a specific server where I can test this?

**Comment by ScionNinja14** [comment]
@sp614x is there a way to private dm the IP? i dont want to release the server publicly

**Comment by ScionNinja14** [comment]
@sp614x 73.176.240.248 ill just delete this message when u join

**Comment by sp614x** [comment]
Looks like a Spigot problem. Maybe the mobs are not initialized properly on the server side or some of their information is missing.

**Comment by Restioson** [comment]
It is not only a spigot problem - I have experienced this on a vanilla sever.

**Comment by mattebad** [comment]
Same here, the server it happened to me on a while back was vanilla, as
well as switching over to my local world which was tested with and without
my resource pack and shader.

**Comment by TboneH** [comment]
It seems to me that it happens most often when I log out near hostile mobs. The hostile mobs that were around when I logged off don't correctly load back in and are invisible. Passive mobs seem unaffected(?). For reference, this is a vanilla 1.13.2 server. Please fix, dying to invisible creepers is really frustrating!

**Comment by Sliminconcoova** [comment]
Although not attacked I noticed something invisible leading two lama's into my base...I didn't record any details before killing what ever it was. I do have bad omen which may attract illusionists which can be invisible?

Looking into my stats, I have killed one wondering trader...which I hadn't realized was a thing. So for me it was that.

**Comment by DataCrusade** [comment]
> Although not attacked I noticed something invisible leading two lama's into my base...I didn't record any details before killing what ever it was. I do have bad omen which may attract illusionists which can be invisible?
> 
> Looking into my stats, I have killed one wondering trader...which I hadn't realized was a thing. So for me it was that.

Yup, Wandering Traders drink invisibility potions at night, no idea why anyone would think to attack them. Illusioners do not spawn in survival, theyre still creative only.

**Comment by sp614x** [comment]
The wandering trader is a 1.14 feature, there is still no OptiFine for 1.14.
@Sliminconcoova Not related to OptiFine.

**Comment by Jxsten** [comment]
Was there ever a fix to this bug..? I am using OptiFine 1.13.2 HD U E7 on a 1.13.2 spigot server. I haven't tested it in vanilla, but this is such an annoying bug.. I cannot hit hostile mobs, or even see them.. I found out that you can light them on fire (I believe that I did anyways.. I wasn't in a controlled environment).

**Comment by ShabanPlayz** [comment]
@sp614x I recently started experiencing this exact issue and it started happening because the server owner of the server i play on increased the entity-tracking-range in the spigot.yml file. Since then I was trying to figure out the cause of the issue and it seems that Optifine is the reason for this happening as this issue does not occur when using a vanilla client.

**Comment by johnvilsack** [comment]
I've been experiencing this in single player.

**Comment by abdking69** [comment]
Hahaha so I have recently killed by Evoker (INVISIBLE Evoker) with no reason It was the first time I noted a BUG and after that It happen again and again with zombies, piglins, zombified piglins, creeper, spider, vindicator and skeletons.
Version: OptiFine 1.16.1 HD U G2

**Comment by abdking69** [comment]
it is happening on even OptiFine 1.16.1 HD U G2!!!!!

**Comment by PossessedRyd** [comment]
So im on 1.16.1 HD U G2 on a fabric-carpet server. And all the invisible mobs ive come across seem to happen when i make a dramatic change in elevation. Ive got a skeleton in a cage fully surrounded with glass and when i relog in the area i can see him fine. However if i fly up about 70 or so blocks and onto a ledge out of sight of the skelly he then becomes invisible when i come back. If i relog this will always fix the issue and any invisible mobs become visible again. For me this is happening within the spawn chunks but it doesnt only happen from my player getting so far away as to deload the entitiy. Getting about three to four  chunks away and out of sight of the mob is what causes mobs to become invisbile. Also it seems to happen more often when im flying working in an area of my world that has a big hole that goes between sea level and bedrock. The unvisibleness only occurs when i travel between the two Y levels and i cant be within a chunk or so of the mob

**Comment by santibag** [comment]
1.16.3 Still happening. Vanilla single player. Happens with or without resource packs.

I noticed that visible mobs may become invisible after death/reload of their chunk.

I didn't notice this on pure vanilla.

**Comment by PhatBoi68419** [comment]
Still happening, I've only experienced this with optifine throughout 1.16, and am currently using 1.16.3. For me it happens most often in the nether, but that could just be because I constantly run into newly generated mobs there and almost never in the overworld.
9/25/20

**Comment by SeaOfPixels** [comment]
Still an issue and is a major one.

**Comment by ShabanPlayz** [comment]
> 
> 
> Still an issue and is a major one.

spx probably will never fix this anyways and you should just switch to sodium anyways since it is a better mod

**Comment by Quavelen** [comment]
> > Still an issue and is a major one.
> 
> spx probably will never fix this anyways and you should just switch to sodium anyways since it is a better mod


Uh, sp614x added this to their 'todo' pile, which is rather large. When they have time to work on issues that are not affecting the majority (we are an extreme minority) then they go through their 'todo' list.

Also, if you read through the first comments on this issue, you'll find that sp614x was unable to reproduce it on their end. They still need someone to give them an infallible way to reproduce it. A while back, I outlined this in an edit to the main issue.

Now regarding Sodium, as nice as its massive performance boost is for potatoes, it lacks many of the features OptiFine offers. Shaders, and CTM being two huge ones for me; neither of which function with Sodium, despite there being Fabric mods for them. Sodium is listed as being incompatible with them.

**Comment by ShabanPlayz** [comment]
> 
> 
> > > Still an issue and is a major one.
> > 
> > 
> > spx probably will never fix this anyways and you should just switch to sodium anyways since it is a better mod
> 
> Uh, sp614x added this to their 'todo' pile, which is rather large. When they have time to work on issues that are not affecting the majority (we are an extreme minority) then they go through their 'todo' list.
> 
> Also, if you read through the first comments on this issue, you'll find that sp614x was unable to reproduce it on their end. They still need someone to give them an infallible way to reproduce it. A while back, I outlined this in an edit to the main issue.
> 
> Now regarding Sodium, as nice as its massive performance boost is for potatoes, it lacks many of the features OptiFine offers. Shaders, and CTM being two huge ones for me; neither of which function with Sodium, despite there being Fabric mods for them. Sodium is listed as being incompatible with them.

Maybe they should make optifine open-source then? I'm sure there is at least one person out there that is capable of reproducing and fixing this bug and has the time to do it.

**Comment by jennasisis** [comment]
> Maybe they should make optifine open-source then? I'm sure there is at least one person out there that is capable of reproducing and fixing this bug and has the time to do it.

@ShabanPlayz Hello, Support member checking in here, ***Optifine will never be open source.*** Taken directly from the discord:

The core of OptiFine consists of many, various changes to Minecraft's rendering code. Rather than simple patches, these are significant reorganizations. This means publishing the full source code of OptiFine would be a direct violation of Minecraft's EULA. Technically, it would be possible to extract the actual changes as patches, which can then be published to GitHub as source code. However, there are a number of issues with this idea. First, OptiFine is built on a custom version of Mod Coder Pack. This non-standard version of MCP is used to allow OptiFine to start development on new versions of the game much, much earlier. During this time, official MCP mappings are either completely missing, or otherwise very unstable. Unfortunately, according to the MCP license and Terms of Usage, modified versions of MCP scripts are not allowed to be distributed. This means that, even if OptiFine patches were released, nobody else would be able to collaborate on the code, defeating the entire purpose of going open-source. Even if somehow all of that was solved, this would still mean significantly changing how OptiFine development is handled. sp614x does not work on the mod using patches, which means he would have to either merge the patches manually, or completely change his entire workflow to use patches. Changing the development process like this would not be a trivial task in the slightest.

In summary, due to a multitude of legal and technical complications, sp614x cannot make OptiFine open-source

**Comment by The-Archives** [comment]
This is a huge, near-gamebreaking issue which has been reported on reddit too. occurs in both the overworld and the nether. please look into this.

**Comment by Vi-Hari** [comment]
> Still happening, I've only experienced this with optifine throughout 1.16, and am currently using 1.16.3. For me it happens most often in the nether, but that could just be because I constantly run into newly generated mobs there and almost never in the overworld.
> 9/25/20

I faced similar issue in nether. Also in overworld sometimes. It especially occurs when entering a newly loaded chunks.

For example, I entered a new bastion in nether after travelling long distance through my nether highway. When I entered it, i could not see any Piglin or Piglin Brutes, but I could hear them and I took severe damage(even in full Netherite armor). 

**I retreated to a safe spot and then quit to main menu and re-entered the world, they are now visible. For players who suffer this issue, this might be a temporary fix.** 

When I reopened the world back there were 4 Piglins and 2 Piglin brutes waiting for me.

**Comment by TomLewis** [comment]
I run a 1.12.2 Paper server with ViaVersion that allows up to 1.16.3 clients to join, and all users that use Optifine experience Invisible entities, Including myself. F3+A shows them again, until they move out of your vision/chunk.
I have tried all versions of Optifine for 1.12.2, they all make entities invisible, I have been trying to solve this for years thinking it was my server at fault.

I have found a way to re-produce this 100% of the time as this invisible entity bug produces another odd bug with blocks for my specific setup. I use a plugin called Movecraft that allows players to build vehicles (ships) out of normal Minecraft blocks they can sail across the sea, this means they can build a ship and sail it, the blocks of their ships get auto moved when they sail! Now I have found that this exact same glitch makes the bocks go invisible when they are updating after sailing for a few seconds. This never happens with a vanilla client.

This is what sailing a ship & invisible blocks look like with optifine:

https://user-images.githubusercontent.com/144046/87859822-f9cf9c00-c92f-11ea-8192-8b165bbb1645.png

What I have noticed with entities like players, is they still make sound and I can see where people are walking from particles near their feet.

@sp614x if there is a way for me to help debug by installing something on my server and testing with players, I can do this.

For my organisation, related ticket https://github.com/eirikh1996/Movecraft/issues/64

**Comment by exapsy** [comment]
This still happens, I'm reporting from Singleplayer, it's not in any way only a multiplayer issue. It's definitely a game breaking bug. :/ But it seems like there are no solutions, no way for anyone to contribute, nor I see any traction or interest from maintainers which is ... sad. :(

**Comment by Siuolthepic** [comment]
Alright, I decided to pull together a current list of known things relating to this glitch.
The current knowledge is:
1. It seems to be a chunk loading thing. 
2. It is more common on Spigot Servers. But has happened in Vanilla Singleplayer+Multiplayer, and Fabric Multiplayer. Its unknown if Fabric SP and Forge Multi or Singleplayer suffer from it but it can be assumed so.
3. It seems pretty random, as recreation of the glitch is pretty inconsistent. 
4. Reports date back to 1.12.2
Along with this, an idea of it being 8 core only problem has been thrown around. While seemingly not objected to, the glitch's sample size might indicate this as false.
I personally recommend people try 1.11 and below to see if this is 1.12.2+ specific or not.

**Comment by ShabanPlayz** [comment]
> 
> 
> Alright, I decided to pull together a current list of known things relating to this glitch.
> The current knowledge is:
> 
>     1. It seems to be a chunk loading thing.
> 
>     2. It is more common on Spigot Servers. But has happened in Vanilla Singleplayer+Multiplayer, and Fabric Multiplayer. Its unknown if Fabric SP and Forge Multi or Singleplayer suffer from it but it can be assumed so.
> 
>     3. It seems pretty random, as recreation of the glitch is pretty inconsistent.
> 
>     4. Reports date back to 1.12.2
>        Along with this, an idea of it being 8 core only problem has been thrown around. While seemingly not objected to, the glitch's sample size might indicate this as false.
>        I personally recommend people try 1.11 and below to see if this is 1.12.2+ specific or not.

I have a dual core processor and I suffer from this glitch

**Comment by Siuolthepic** [comment]
By the way, has there been ANY reports about this bug on 1.14.4?

**Comment by BobBobilou** [comment]
This really need to be fix because it almost made 1 years since each time I upgrade to an new version of optifine, almost each world had this bug. I play hardcore so there nothing more frustrating than get kill by an invisible mod.
I use 1.16.4 HD_U_G5.jar
you can it the invisible mod by punching with an sword an mod next to it
can burn the mod by lighting the block below him.
Please fix this bug it break the game a lot and I love optifine so I dont want to have to play minecraft without any mod.

**Comment by Carlifoxy** [comment]
Is there any solution ? I have a harcore world and I had to stop using optifine so that the invisible entities wouldn't kill me, it's not pretty to play at 20fps because I don't feel safe using optifine

**Comment by ShabanPlayz** [comment]
> 
> 
> Is there any solution ? I have a harcore world and I had to stop using optifine so that the invisible entities wouldn't kill me, it's not pretty to play at 20fps because I don't feel safe using optifine

install sodium, lithium and phosphor

**Comment by berkkrkc09** [comment]
This is happenning to me on 1.16.4(with forge) as well. But I play with around 200 mods and it takes 5 minutes every time I launch the game, plus loading worlds take 1 to 2 minutes too. People keep telling to restart the game. Restarting is not a fix it is a workaround and this needs to get fixed. All mods that optimise the lighting engine(Sodium, Lithium, Phosphor...) are for fabric so there is no other way but waiting for Forge users.

**Comment by elman879** [comment]
I had this problem a while back and uninstalled Optifine, here was my F3 screen at the time: https://imgur.com/a/1xgGtGq 
I opened an issue (#4922) because I saw that this had been closed when it was still occurring in my 1.16.3 game. You can see some more information there. The invisible mobs would reappear if you save & exit and reenter the game. Optifine is my only mod, though at one point I did have shaders (BSL Shaders I think) but I don't think they affected anything. Let me know if there's any more information I can put here to help fix this, I'd like to reinstall Optifine if this gets fixed. I don't remember my specific settings I had in Optifine at the time since this was several months ago.

**Comment by SkyeMors** [comment]
Looks like this is (a?) main thread for this.

Encountering this (invisible hostile mobs) in MC 1.16.5 with OptiFine_1.16.5_HD_U_G6 and Forge 36.0.0.
Walking around, hear Hmm and get shot at, nothing nearby and seems there's 2-3 wandering Pillagers that are active but are otherwise unloaded. No model, no shadow, still show up on Xaero's minimap. Reloading world temporarily fixes this.

**Comment by ZenIsBestWolf** [comment]
We're now treating #5016 as the main issue, as it has become much more focused around 1.16. Please direct all relevant conversation about it to that issue.

**Comment by Tom28281** [comment]
still happening in 1.12 version HD_U_G5

**Comment by TomLewis** [comment]
I ended up moving to fabulously optimised, they won't fix this.
